### PR TITLE
 [NCL-7207] Implement Insert / Remove dependencies functionality for Build Config

### PR DESCRIPTION
**Pull request by patrikk0123**

### CODE REVIEW

File: src/components/BuildConfigDependenciesEditPage/BuildConfigDependenciesEditPage.tsx [hunk 1]
```diff
@@ -0,0 +1,261 @@
...
+  const serviceContainerBuildConfig = useServiceContainer(buildConfigApi.getBuildConfig);
+  const serviceContainerBuildConfigRunner = serviceContainerBuildConfig.run;
+
+  const serviceContainerBuildConfigDependencies = useServiceContainer(buildConfigApi.getDependencies);
+  const serviceContainerBuildConfigDependenciesRunner = serviceContainerBuildConfigDependencies.run;
+
+  const serviceContainerBuildConfigs = useServiceContainer(buildConfigApi.getBuildConfigs);
+  const serviceContainerBuildConfigsRunner = serviceContainerBuildConfigs.run;
+
+  const serviceContainerBuildConfigEdit = useServiceContainer(buildConfigApi.editBuildConfig, 0);
```
[the remark this answers is not in the document]
Comment (patrikk0123) [reply src/components/BuildConfigDependenciesEditPage/BuildConfigDependenciesEditPage.tsx:49]: I would use patch, but REST API supports only put in this case.

[the remark this answers is not in the document]
Comment (matedo1) [reply src/components/BuildConfigDependenciesEditPage/BuildConfigDependenciesEditPage.tsx:49]: We agreed to investigate it more deeply, according to swagger it should be supported, possibly two follow up jiras will be created (one for backend to fix it, one for UI to integrate it)

[the remark this answers is not in the document]
Comment (patrikk0123) [reply src/components/BuildConfigDependenciesEditPage/BuildConfigDependenciesEditPage.tsx:49]: It was my fault, I was sending the backend the wrong data.

**Comment by matedo1** [comment]
I tested it locally and write jira comment with questions

**Comment by matedo1** [comment]
I tested it locally and write jira comment with issues

**Review by matedo1 — CHANGES_REQUESTED** [review]
I tested it locally and write jira comment with issues

File: src/components/BuildConfigDependenciesEditPage/BuildConfigDependenciesEditPage.tsx [hunk 2]
```diff
@@ -0,0 +1,261 @@
...
+
+import { useParamsRequired } from 'hooks/useParamsRequired';
+import { usePatchOperation } from 'hooks/usePatchOperation';
+import { useQueryParamsEffect } from 'hooks/useQueryParamsEffect';
+import { useServiceContainer } from 'hooks/useServiceContainer';
+import { useTitle } from 'hooks/useTitle';
+
+import { ActionConfirmModal } from 'components/ActionConfirmModal/ActionConfirmModal';
+import { PageLayout } from 'components/PageLayout/PageLayout';
+import { ConfigsAddList } from 'components/ProductVersionBuildConfigsEditPage/ConfigsAddList';
```
[the remark this answers is not in the document]
Comment (matedo1) [reply src/components/BuildConfigDependenciesEditPage/BuildConfigDependenciesEditPage.tsx:17]: ping

File: src/components/ProductVersionBuildConfigsEditPage/ConfigsAddList.tsx [hunk 3]
```diff
@@ -80,6 +86,12 @@ export const ConfigsAddList = <T extends BuildConfiguration | GroupConfiguration
     (config: T) =>
       addedConfigs.every((addedConfig) => addedConfig.id !== config.id) &&
       (!productVersionToExclude || config.productVersion?.id !== productVersionToExclude) &&
+      (!buildConfigToExclude || !isBuildConfig(config) || config.id !== buildConfigToExclude) &&
```
Comment (matedo1) [inline src/components/ProductVersionBuildConfigsEditPage/ConfigsAddList.tsx:89]: Please create some named variable explaining individual condition groups, or at least some TSDocs would be helpful. Now it's very hard to maintain.

**Review by matedo1 — CHANGES_REQUESTED** [review]
I successfully tested it locally. Thanks

File: src/components/BuildConfigDependenciesEditPage/BuildConfigDependenciesEditPage.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (patrikk0123) [reply src/components/BuildConfigDependenciesEditPage/BuildConfigDependenciesEditPage.tsx:17]: I don't know what you mean by this comment.

**Comment by matedo1** [comment]
Again, I successfully tested it locally. Thanks

**Review by matedo1 — APPROVED** [review]

**Pull request merged**